src/session_impl.cpp

```diff
@@ -1876,6 +1892,43 @@ namespace {
 			interface_to_endpoints(device, port, ssl, eps);
 		}
 
+#if TORRENT_USE_IPV6
+		std::vector<ip_interface> const ifs = enum_net_interfaces(m_io_service, ec);
+		if (!ec)
+		{
+			// if any listen endpoints are on [::] remove them and add endpoints for
+			// each interface that isn't already included in eps
```

Comment:
it looks like this would be a good candidate for factoring out into a free function and unit test